src/vm/comdelegate.cpp

```diff
@@ -2931,7 +2931,58 @@ PCODE COMDelegate::GetSecureInvoke(MethodDesc* pMD)
 #ifdef FEATURE_CAS_POLICY
 #error GetSecureInvoke not implemented
 #else
-    UNREACHABLE();
+    GCX_PREEMP();
+
+    MetaSig sig(pMD);
+
+    BOOL fReturnVal = !sig.IsReturnTypeVoid();
+
+    SigTypeContext emptyContext;
+    ILStubLinker sl(pMD->GetModule(), pMD->GetSignature(), &emptyContext, pMD, TRUE, TRUE, FALSE);
+
+    ILCodeStream *pCode = sl.NewCodeStream(ILStubLinker::kDispatch);
+
+    DWORD dwReturnValNum = -1;
```

Comment:
This special handling of return values should not be needed here. If you delete everything that has to do with `dwReturnValNum` and `fReturnVal`, everything should still work fine.